src/widgets/scrollbar.rs

```diff
@@ -67,35 +120,79 @@ impl<'a> Scrollbar<'a> {
 
     /// Sets the orientation of the scrollbar.
     pub fn orientation(mut self, orientation: Orientation) -> Self {
-        self.orientation = orientation;
-        self.bar_symbol = match self.orientation {
+        self.orientation = orientation.clone();
+        self.symbol(match orientation {
             Orientation::Vertical => DOUBLE_VERTICAL,
             Orientation::Horizontal => DOUBLE_HORIZONTAL,
-        };
-        self
+        })
     }
 
     /// Sets the symbol that represents the scroller.
-    pub fn symbol(mut self, symbol: &'a str) -> Self {
-        self.symbol = symbol;
+    pub fn thumb_symbol(mut self, thumb_symbol: &'a str) -> Self {
+        self.thumb_symbol = thumb_symbol;
         self
     }
 
     /// Sets the style that is used for the scroller.
-    pub fn style(mut self, style: Style) -> Self {
-        self.style = style;
+    pub fn thumb_style(mut self, thumb_style: Style) -> Self {
+        self.thumb_style = thumb_style;
         self
     }
 
     /// Sets the symbol that represents the bar of the scrollbar.
-    pub fn bar_symbol(mut self, bar_symbol: &'a str) -> Self {
-        self.bar_symbol = bar_symbol;
+    pub fn track_symbol(mut self, track_symbol: &'a str) -> Self {
+        self.track_symbol = track_symbol;
         self
     }
 
     /// Sets the style that is used for the bar of the scrollbar.
-    pub fn bar_style(mut self, bar_style: Style) -> Self {
-        self.bar_style = bar_style;
+    pub fn track_style(mut self, track_style: Style) -> Self {
+        self.track_style = track_style;
+        self
+    }
+
+    /// Sets the symbol that represents the beginning of the scrollbar.
+    pub fn begin_symbol(mut self, begin_symbol: &'a str) -> Self {
+        self.begin_symbol = begin_symbol;
+        self
+    }
+
+    /// Sets the style that is used for the beginning of the scrollbar.
+    pub fn begin_style(mut self, begin_style: Style) -> Self {
+        self.begin_style = begin_style;
+        self
+    }
+
+    /// Sets the symbol that represents the end of the scrollbar.
+    pub fn end_symbol(mut self, end_symbol: &'a str) -> Self {
+        self.end_symbol = end_symbol;
+        self
+    }
+
+    /// Sets the style that is used for the end of the scrollbar.
+    pub fn end_style(mut self, end_style: Style) -> Self {
+        self.end_style = end_style;
+        self
+    }
+
+    pub fn symbol(mut self, symbol: Set) -> Self {
+        self.track_symbol = symbol.track;
+        self.thumb_symbol = symbol.thumb;
+        self.begin_symbol = symbol.begin;
+        self.end_symbol = symbol.end;
+        self
+    }
+
+    pub fn style(mut self, style: Style) -> Self {
+        self.track_style = style.clone();
+        self.thumb_style = style.clone();
+        self.begin_style = style.clone();
+        self.end_style = style.clone();
+        self
+    }
+
+    pub fn arrows(mut self, arrows: bool) -> Self {
+        self.arrows = arrows;
         self
     }
```

Comment:
Do you think we could omit this and make start and end `Optional<&str>` where there's nothing rendered if they're set to none?

Reply:
Do we want to support arrows just at the beginning or just at the end, if one is set to `Some` and the other is set to `None`? The rendering logic gets more complicated in that case.

Reply:
Perhaps we could have an `arrows(start: &str, end: &str)` method?

Reply:
I've added a version in https://github.com/tui-rs-revival/ratatui/pull/228/commits/aec39611820f4cbbea501fbcbad498444d2b3790 and https://github.com/tui-rs-revival/ratatui/pull/228/commits/e6a68954245ccf2eee1c9f7354dc195b77e0a49f that uses `Option<&str>` for the begin and end symbols.

Reply:
I believe this comment has been addressed now.